.github/workflows/Tests.yml

```diff
@@ -172,99 +166,15 @@ jobs:
-      run: |
-        pip install --upgrade "pip>=21.0" "setuptools>=45.0" "wheel>=0.37" "build>=1.0.1";
-        pip install -r ./requirements.txt ;
-        pip install -r ./tests/requirements.txt || true ;
-        pip install --upgrade -r ./docs/requirements.txt || true ;
-    - name: Install code-climate tools for ${{ matrix.python-version }} on ${{ matrix.os }}
-      if: ${{ !cancelled() && runner.os == 'Linux' }}
-      shell: bash
-      run: |
-        curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter || true ;
-        chmod +x ./cc-test-reporter 2>/dev/null || true ;
-        ./cc-test-reporter before-build || true ;
-    - name: Install deepsource tools for ${{ matrix.python-version }} on ${{ matrix.os }}
-      if: ${{ !cancelled() && runner.os == 'Linux' }}
-      shell: bash
-      run: |
-        (curl https://deepsource.io/cli | sh) || true ;
-    - name: Pre-Clean
-      id: clean
-      run: make -j1 -f Makefile clean || true ;
-    - name: Generate Coverage for py3.9 on ${{ matrix.os }}
-      if: ${{ runner.python-version == '3.9' }}
-      run: make -f Makefile test ;
-    - name: Generate Coverage for py${{ matrix.python-version }} on ${{ matrix.os }}
-      if: ${{ runner.python-version != '3.9' }}
-      run: make -f Makefile test-pytest ;
-    - name: Upload Python ${{ matrix.python-version }} coverage to Codecov
-      uses: codecov/codecov-action@v4
-      with:
-        token: ${{ secrets.CODECOV_TOKEN }}
-        files: ./coverage.xml
-        directory: .
-        flags: multicast,${{ matrix.os }},${{ matrix.python-version }}
-        name: multicast-github-${{ matrix.os }}-${{ matrix.python-version }}
-        verbose: true
-        fail_ci_if_error: false
-    - name: Upload Python ${{ matrix.python-version }} Artifact
-      uses: actions/upload-artifact@v4
-      with:
-        name: Test-Report-${{ matrix.os }}-${{ matrix.python-version }}
-        path: ./test-reports/
-        if-no-files-found: ignore
-    - name: code-climate for ${{ matrix.python-version }}
-      if: ${{ !cancelled() && runner.os == 'Linux' }}
-      shell: bash
-      run: |
-        ./cc-test-reporter after-build --exit-code 0 || true ;
-    - name: deepsource for ${{ matrix.python-version }}
-      if: ${{ !cancelled() && runner.os == 'Linux' }}
-      shell: bash
-      run: |
-        ./bin/deepsource report --analyzer test-coverage --key python --value-file ./coverage.xml 2>/dev/null || true ;
-    - name: Post-Clean
-      id: post
-      run: make -j1 -f Makefile clean || true ;
-      if: ${{ always() }}
-
-  COVERAGE:
-    if: ${{ success() }}
-    needs: [BUILD, MATS, COVERAGE-MATS]
-    runs-on: ${{ matrix.os }}
-    timeout-minutes: 10
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-13, windows-latest]
-        python-version: [3.7, 3.8, 3.9, "3.10"]
+        python-version: ["3.10", "3.11", "3.12"]
```

Comment:
LGTM! We are now dropping support for pre-3.9 (as part of #134) Makes sense to consolidate versions.